### ISSUE TITLE
[1.1.9] Undemanded Diagonal Moves Across Corners

### ISSUE THREAD

**Issue by RobinXe** [MarlinFirmware/Marlin#11906]
During printing of a part with several right-angles (possibly irrelevant), the printer appears to sum the demanded vectors of several XY travels, and proceeds diagonally, across the corner.

These diagonal translations do not appear in the gcode, either as extrusions, or as travels, so it is not a slicing/stringing issue. The omitted translations are visible as gaps in the outside wall.

![fb_img_1537782393286](https://user-images.githubusercontent.com/43536018/45946559-24c55f00-bfe9-11e8-9326-0abac8b00898.jpg)

Using Tevo Tornado running MKS Gen L board, with stock drivers (vrefs adjusted), Marlin 1.1.9 [Marlin Conf Files.zip](https://github.com/MarlinFirmware/Marlin/files/2410406/Marlin.Conf.Files.zip), Cura 3.4.1, BLTouch and Bilenear ABL, printing via USB.

When trying to print the parts from [here](https://www.thingiverse.com/thing:2781135), using the attached gcode [TT_Tornado_Ultrabase_Clamp_Default_-_Left.zip](https://github.com/MarlinFirmware/Marlin/files/2410400/TT_Tornado_Ultrabase_Clamp_Default_-_Left.zip) these diagonal translations repeatedly occur at various, non-consistent, layers throughout the print.

I have observed these moves happening, after a slight pause, and on occasion they occur through the interior of the print as well. Printing at a slower speed does not rectify the issue. The rest of the prints complete acceptably without any layer-shift, so positional awareness is not being lost.

My initial thoughts are that it could be the rx buffer is filling up, and Marlin recovering space by combining several moves into one, though I don't know if this behaviour is intended.

**Comment by gloomyandy** [comment]
Do you have an SD card reader? What happens if you print from the SD card rather than via an external program?

**Comment by RobinXe** [comment]
That's the next thing to explore. Waiting on delivery of an SD card.

**Comment by ManuelMcLure** [comment]
Are you actually observing the head moving diagonally or only the resulting diagonal strings? Because if for some reason the filament didn't stick to the layer below the result would be very similar even if the head moved correctly. A slight extra increase in Z would result in this, and I see possible evidence of that in the front edge of the piece.

**Comment by RobinXe** [comment]
Yes, I've watched it; there seems to be the slightest pause, then the tone of the steppers changes notably, and it makes a purposeful extrusion across the corner.

The Z steps you see on the front are where it has done the same diagonal translation across the inside of the piece.

**Comment by tcm0116** [comment]
Is the issue repeatable in that you'll get the same diagonal lines if you print the same object a second time? If so, it might be helpful to try and narrow it down to a small subset of the GCode that's exhibiting the issue. 

If you're using a host, such as OctoPrint, it might be a good idea to check the print log to see if there were any errors reported while sending the GCode to the printer.

**Comment by thinkyhead** [comment]
Just for fun, see if setting `MIN_STEPS_PER_SEGMENT` to 0 makes any difference.

**Comment by thinkyhead** [comment]
Are you using a Windows machine by chance? There are hidden options in the Device Manager to tune the system serial port so that it uses a smaller buffer. It's been described in another issue, but I can't find it at the moment.

**Comment by GMagician** [comment]
> There are hidden options in the Device Manager to tune the system serial port so that it uses a smaller buffer. 

Forgive my ignorance, but why a small buffer from windows side may change something in profile?

**Comment by boelle** [comment]
@RobinXe is problem gone in latest bugfix 2.0?

**Comment by boelle** [comment]
@thinkyhead i think we can close this one

**Comment by thinkyhead** [comment]
@GMagician — Something related to latency. I don’t know more.

**Comment by RobinXe** [comment]
I'm sorry, for some reason I've missed most of the replies to this.

I haven't tried to remedy the issue any further, I've just been printing from SD card.

Happy to close (not that it's up to me!).
